### PR TITLE
WIP libspatialite

### PR DESCRIPTION
**Pull request by ocefpaf**
Another optional dependency for `gdal`.

### CODE REVIEW

File: recipes/libspatialite/build.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+export LDFLAGS=-L$PREFIX/lib
+export CFLAGS=-I$PREFIX/include
```
Comment (jakirkham) [inline recipes/libspatialite/build.sh]: Why do you need these? Does it not pick these up anyways?

**Comment by jakirkham** [comment]
One minor comment. Otherwise LGTM.

File: recipes/libspatialite/build.sh (hunk 1 above)
Comment (ocefpaf) [inline recipes/libspatialite/build.sh]: Without those I get:

```shell
# configure: error: cannot find sqlite3.h, bailing out
# Command failed: /bin/bash -x -e /home/build.sh
```

**Comment by jakirkham** [comment]
Sorry, on my phone, is there an option to tell configure where SQLite is?

**Comment by pelson** [comment]
On osx:

```
/bin/sh ../libtool  --tag=CC   --mode=link gcc  -I/Users/travis/miniconda/envs/_build/include -version-info 8:0:1 -L/Users/travis/miniconda/envs/_build/lib -o libspatialite.la -rpath /Users/travis/miniconda/envs/_build/lib versioninfo/version.lo ./gaiaaux/libgaiaaux.la ./gaiaexif/libgaiaexif.la ./gaiageo/libgaiageo.la ./geopackage/libgeopackage.la ./spatialite/libsplite.la ./shapefiles/libshapefiles.la ./dxf/libdxf.la ./md5/libmd5.la ./control_points/libcontrol_points.la ./srsinit/libsrsinit.la ./connection_cache/libconnection_cache.la ./virtualtext/libvirtualtext.la ./wfs/libwfs.la -L/Users/travis/miniconda/envs/_build/lib -lxml2   -lpthread -ldl -lm -lfreexl -liconv -lproj -lz -lsqlite3  -L/Users/travis/miniconda/envs/_build/lib -lgeos_c
grep: /usr/lib/libiconv.la: No such file or directory
sed: /usr/lib/libiconv.la: No such file or directory
libtool: link: `/usr/lib/libiconv.la' is not a valid libtool archive
make[3]: *** [libspatialite.la] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
Command failed: /bin/bash -x -e /Users/travis/build/conda-forge/staged-recipes/recipes/libspatialite/build.sh
```

Another case of bad homebrew?

**Comment by ocefpaf** [comment]
**Do not merge* this!!! I just want to test what will happens if I remove the `/usr/local`*

**Comment by ocefpaf** [comment]
> Sorry, on my phone. Is there an option to tell configure where SQLite is?

I could not find one.

**Comment by ocefpaf** [comment]
```shell
+ rm -rf /usr/local
rm: /usr/local: Permission denied
```

Forgot that we are not in our docker image :smile: 

[I guess](https://github.com/conda-forge/staged-recipes/blob/master/.travis.yml#L4) we'll have to brew uninstall stuff :unamused:

**Comment by pelson** [comment]
How about:

```
brew remove --force $(brew list)
```

http://blog.zedroot.org/uninstall-all-the-installed-packages-with-homebrew/

**Comment by ocefpaf** [comment]
That worked! And exposed another problem. Even though I installed `libxml2` from conda and added the path in the configure option we were getting the homebrew version before!

```
configure: error: 'libxml2' is required but it doesn't seem to be installed on this system.
Command failed: /bin/bash -x -e /Users/travis/build/conda-forge/staged-recipes/recipes/libspatialite/build.sh
```

**Comment by pelson** [comment]
> That worked! And exposed another problem. Even though I installed libxml2 from conda and added the path in the configure option we were getting the homebrew version before!

Awesome! That is why we CI this stuff! :+1:

**Comment by ocefpaf** [comment]
Getting on Travis-CI:

```shell
gcc -DHAVE_CONFIG_H -I. -I..  -I/Users/travis/miniconda/envs/_build/include -I/Users/travis/miniconda/envs/_build/include   -I./../src/headers -I. -I/Users/travis/miniconda/envs/_build/include -MT check_endian.o -MD -MP -MF .deps/check_endian.Tpo -c -o check_endian.o check_endian.c
mv -f .deps/check_endian.Tpo .deps/check_endian.Po
/bin/sh ../libtool  --tag=CC   --mode=link gcc -I./../src/headers -I. -I/Users/travis/miniconda/envs/_build/include -L../src -lpthread -lspatialite -lm -lxml2  -L/Users/travis/miniconda/envs/_build/lib -o check_endian check_endian.o  -lfreexl -lcharset -liconv -lproj -lz -lsqlite3  -L/Users/travis/miniconda/envs/_build/lib -lgeos_c
libtool: link: cannot find the library `/usr/lib/libiconv.la' or unhandled argument `/usr/lib/libiconv.la'
make[3]: *** [check_endian] Error 1
make[2]: *** [check-am] Error 2
make[1]: *** [check-recursive] Error 1
make: *** [check-recursive] Error 1
Command failed: /bin/bash -x -e /Users/travis/build/conda-forge/staged-recipes/recipes/libspatialite/build.sh
```

**Comment by pelson** [comment]
Are you expecting libiconv to be pulled in by something?

**Comment by pelson** [comment]
i.e. if you add libiconv into the requirements, does it solve the problem :smile:

**Comment by ocefpaf** [comment]
I did that already. (Re-basing masked that, sorry.)

**Comment by jakirkham** [comment]
It might be that one of the dependencies is linking against a libiconv from the system instead of the one in conda-forge.

**Comment by ocefpaf** [comment]
> It might be that one of the dependencies is linking against a libiconv from the system instead of the one in conda-forge.

If that is the case we should re-build `geos`, `proj.4`, and `freexl`. I am tempted to skip OS X for now...

**Comment by jakirkham** [comment]
Shouldn't be difficult. We just try bumping the build numbers on those with this brew clean step and see how well it goes. If all goes well, we have new binaries that are definitely clean. If not, we can find what the problems are.

**Comment by ocefpaf** [comment]
> Shouldn't be difficult. 

Difficult? No. We already have https://github.com/conda-forge/conda-smithy/pull/114. It is just a matter of time now...

**Comment by jakirkham** [comment]
Did we add it to staged-recipes too? Sorry, I have lost track if we did.

**Comment by ocefpaf** [comment]
Nope. This PR was the first one to test it. (And I just removed and re-based. I am still thinking if we should merge this and get `gdal` going or if it is worth waiting for the re-builds.)

**Comment by jakirkham** [comment]
Honestly, no preference. I leave it up to you. If you really need `gdal` now, we should get it in. If not, maybe we should wait for the rebuilds.

**Comment by ocefpaf** [comment]
We do have a broken `gdal` (and `fiona`), but this is not part of the fix. I just wanted to put this in there to avoid another re-build. Maybe I should just fix `gdal` and leave this to another day.

**Comment by jakirkham** [comment]
Sorry, meant `libspatialite`.

**Comment by ocefpaf** [comment]
> Sorry, meant `libspatialite`.

Same answer though :smile:

**Comment by jakirkham** [comment]
Well, confusion aside, I'm :+1: either way.

File: .CI/build_all [hunk 2]
```diff
@@ -7,6 +7,9 @@ export CONDA_NPY=19
 # We only build if we aren't a merged PR (that is, only build for PRs).
 if [ -z "$GH_TOKEN" ]; then
 
+    # Remove homebrew.
+    brew remove --force $(brew list)
```
Comment (ocefpaf) [inline .CI/build_all]: @jakirkham and @pelson is this OK?

**Comment by ocefpaf** [comment]
Using `libgcc` for

```
libgcc-4.8.5-1:
    libgcc_s.so.1 (lib/libgcc_s.so.1)
    libstdc++.so.6 (lib/libstdc++.so.6)
```

This might not be a good idea...

**Comment by ocefpaf** [comment]
I am getting 3 failures on OS X now. (Better than 84 :wink:)

```
FAIL: check_extension
FAIL: check_sql_stmt
FAIL: check_get_normal_zoom_extension_load
```

I will skip the tests on OS X and open an issue for this on the feedstock.

**Pull request merged**